tests/e2eTest/v0.3.4.ts

```diff
@@ -290,20 +295,164 @@ const test_suit = (setTezosToolkit: (tezos: TezosToolkit) => TezosToolkit) =>
           });
 
         after_storage.proposals
-          .get(after_proposal_counter.toNumber() - 1)
+          .get(after_proposal_counter.toNumber() - 2)
           .then((value: proposal) => {
             expect(value.contents.length).toBe(1);
             expect(
               value.contents.find(v => "proof_of_event" in v)
             ).toBeDefined();
           });
+      },
+      { timeout: tenMins }
+    );
+
+    it(
+      "test submitTxProposals with sign",
+      async () => {
+        const v = VersionedApi(version, addr);
+        const tzsafe = await retry(() => tezos.wallet.at(addr));
+        const proposals: proposals = {
+          transfers: [
+            {
+              type: "transfer",
+              values: {
+                to: owner,
+                amount: "1",
+              },
+            },
+            {
+              type: "transfer",
+              values: {
+                to: owner,
+                amount: "2",
+              },
+            },
+            {
+              type: "poe",
+              values: {
+                payload: "0001",
+              },
+              fields: [],
+            },
+          ],
+        };
+
+        const before_storage: contractStorage = await retry(() =>
+          tzsafe.storage()
+        );
+        const before_proposal_counter: BigNumber =
+          before_storage.proposal_counter;
+
+        const deployed_result = await retry(() =>
+          v.submitTxProposals(
+            tzsafe,
+            tezos,
+            proposals,
+            true,
+            undefined,
+            true,
+            false
+          )
+        );
+
+        const [_, msg] = deployed_result;
+        expect(msg.startsWith("op")).toBeDefined();
+
+        // proposal validation
+        const after_storage: contractStorage = await retry(() =>
+          tzsafe.storage()
+        );
+
+        const after_proposal_counter: BigNumber =
+          after_storage.proposal_counter;
+
+        expect(after_proposal_counter.toNumber()).toBe(
+          before_proposal_counter.toNumber() + 2
+        );
         after_storage.proposals
-          .get(after_proposal_counter.toNumber() - 2)
-          .then((value: proposal) => {
-            expect(value.contents.filter(v => "transfer" in v).length).toBe(2);
+          .get(after_proposal_counter.toNumber())
+          .then((value: any) => {
+            expect(value).toBeDefined();
+            expect(value.contents.length).toBe(2);
           });
+
+        // sign validation
+        const after_proposal: any = await retry(() =>
+          after_storage.proposals.get(after_storage.proposal_counter)
+        );
+        const after_sigs: MichelsonMap<string, boolean> =
+          after_proposal.signatures;
+        expect(after_sigs.size).toBe(1);
+        expect(after_sigs.get(owner)).toBe(true);
```

Comment:
let's also check poe proposal is signed.
```suggestion
        expect(after_sigs.get(owner)).toBe(true);
        
        const after_poe_proposal: any = await retry(() =>
          after_storage.proposals.get(after_storage.proposal_counter.minus(1))
        );
        const after_poe_sigs: MichelsonMap<string, boolean> =
          after_poe_proposal.signatures;
        expect(after_poe_sigs.size).toBe(1);
        expect(after_poe_sigs.get(owner)).toBe(true);
```